### DISCUSSION TITLE
Remote: Repository not found. fatal: repository https://github... not found

### DISCUSSION THREAD

**Discussion by mariokart229** [community/community#26896] [Actions]
<p>Hi,<br>
I am having problems creating a git repository automatically, using powershell and git. Can this be done? Could you help me?<br>
Here is what I am doing:</p>
<pre><code class="lang-auto">mkdir &lt;new directory&gt;
cd &lt;new directory&gt;
git init
&lt;drop a file into new directory&gt;
git add . 
git commit -m "Autogenerated repo"
$url = "https://$pat@github.com/$user/$repo.git"
git remote add origin $url
git remote set-url origin $url
git push -u origin master 
</code></pre>
<p>I get the error:<br>
<strong>remote: Repository not found.<br>
fatal: repository ‘<a href="https://github" rel="noopener nofollow ugc">https://github</a>…’ not found</strong></p>
<p>BTW, PAT has full repo access rights and can upload code into existing repo just fine.<br>
Regards,<br>
Mario C.</p>

**Comment by airtower-luna** [comment]
<p>The repository must exist on Github before you can push to it. If you’d like to automate that you should be able to do that <a href="https://docs.github.com/en/rest/reference/repos#create-a-repository-for-the-authenticated-user">using the API</a>. As far as I know there are tools that wrap that API in a command line or library for you, but I’m not familiar with them.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mariokart229">@mariokart229</a> ,</p>
<p>As <a class="mention" href="/airtower-luna">@airtower-luna</a> mentioned, before you can push to the remote repository, this remote repository must have been created in the specified organization account or user account, and generally this remote repository is empty that does not contains any files.<br>
For example, on the web page, an empty remote repository looks like this:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/19a248e87cbed4c36974fd336c7745264d42fd00.png" data-download-href="/uploads/short-url/3ELD50MvYyW9jjbNtDHYGro8jyo.png?dl=1" title="empty-repo" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181100502-734c715f-dd14-4b44-99a5-181ac8c6574f.png" alt="empty-repo" data-base62-sha1="3ELD50MvYyW9jjbNtDHYGro8jyo" width="690" height="414" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/19a248e87cbed4c36974fd336c7745264d42fd00_2_690x414.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/19a248e87cbed4c36974fd336c7745264d42fd00_2_1035x621.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/1/19a248e87cbed4c36974fd336c7745264d42fd00_2_1380x828.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/19a248e87cbed4c36974fd336c7745264d42fd00_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">empty-repo</span><span class="informations">1558×937 63.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If you want to use a shell script to automatically create the empty remote repository and push some files to it, instead of manually creating the repository. You can run the “<a href="https://docs.github.com/en/rest/reference/repos#create-an-organization-repository" rel="noopener nofollow ugc"><strong>Create an organization repository</strong></a>” or “<a href="https://docs.github.com/en/rest/reference/repos#create-a-repository-for-the-authenticated-user" rel="noopener nofollow ugc"><strong>Create a repository for the authenticated user</strong></a>” API at first in the script.</p>

**Comment by mariokart229** [comment]
<p>Hi brightran,<br>
thank you for your reply. I do not see an empty repository in your supplied image. The image has the same code for creating the repo as the one I have tried, with the exception of the additional “git branch -M master”. I have tried adding it. Results are the same. As you and airtower-luna suggest, API could work. I have yet to see it do so.</p>

**Comment by mariokart229** [comment]
<p>Thank you for your reply. I am trying to get it to work via the API. Here is what I am trying:</p>
<pre><code class="lang-auto">    curl.exe -H @{"Authorization"="token $GHPAT"} https://github.com/api/v3/users/$GHUser/repos `
             -d @{"name"="$GHRepo"}
</code></pre>
<p>Results in:<br>
<em>&gt; curl.exe :   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current</em><br>
<em>&gt; At C:\Users\extcloft\Desktop\Azure To Git Backup\migrate.ps1:39 char:5</em><br>
<em>&gt; +     curl.exe -H @{“Authorization”=“token $GHPAT”} <a href="https://github.com/" rel="noopener nofollow ugc">https://github.com/</a> …</em><br>
*&gt; +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*<br>
<em>&gt;     + CategoryInfo          : NotSpecified: (  % Total    % …  Time  Current:String) [], RemoteException</em><br>
<em>&gt;     + FullyQualifiedErrorId : NativeCommandError</em></p>
<p>No repo is created.</p>
<p>Other stuff tried, yielding same or other errors:</p>
<p><code>curl.exe -u ($GHUser+":"+$GHPAT) https://github.com/api/v3/users/$GHUser/repos -d "name"="$GHRepo"</code></p>
<p><code>curl -H @{"Authorization"="token $GHPAT"} https://github.com/api/v3/users/$GHUser/repos -d @{"name"="$GHRepo"}</code></p>
<pre><code class="lang-auto">$CurlArgument = '-H', "@{'Authorization'='token $GHPAT'}", 
                    '-X', 'POST',
                    'https://github.com/api/v3/users/$GHUser/repos',
                    '-d', '@{"name"="$GHRepo"}'
    $CURLEXE = 'C:\Program Files\Git\mingw64\bin\curl.exe'
    &amp; $CURLEXE @CurlArgument
</code></pre>
<pre><code class="lang-auto">$form = @{
        H  = @{'Authorization'='token $GHPAT'}
       d  = @{"name"="$GHRepo"}
 }
 $result = Invoke-WebRequest -Uri 'https://github.com/api/v3/users/$GHUser/repos' -Method Post -Body $form
</code></pre>

**Comment by mariokart229** [comment, marked as answer]
<p>Found solution</p>
<pre><code class="lang-auto">    $authHeader= New-Object 'System.Collections.Generic.Dictionary[String,String]'
    $authHeader.Add("Authorization", "token $GHPAT")

    #list repos
    #curl -H $authHeader -uri https://api.github.com/user/repos
    
    $body = '{"name":"'+"$GHRepo"+'"}'
    $reply = Invoke-RestMethod -H $authHeader -uri https://api.github.com/user/repos -Method Post -body $body
</code></pre>
